src/main/java/org/apache/sling/feature/maven/mojos/ApisJarMojo.java

```diff
@@ -337,25 +357,92 @@ private void downloadSources(ArtifactId artifactId, File deflatedSourcesDir, Fil
 
                 Scm scm = pomModel.getScm();
                 if (scm == null) {
-                    throw new MojoExecutionException("SCM not defined in POM " + pomArtifactId + ", sources can not be retrieved");
+                    if (suppressSCMResolutions != null && suppressSCMResolutions.contains(pomArtifactId.toMvnId())) {
+                        getLog().warn("SCM not defined in POM " + pomArtifactId + ", sources can not be retrieved, but ignored due to the suppressSCMResolutions configuration");
+                        return;
+                    } else {
+                        throw new MojoExecutionException("SCM not defined in POM " + pomArtifactId + ", sources can not be retrieved");
+                    }
+                }
+
+                String connection = scm.getConnection();
+                String tag = scm.getTag();
+
+                if (scmRewrite != null && !scmRewrite.isEmpty()) {
+                    dance : for (Entry<Object, Object> rewrite : scmRewrite.entrySet()) {
+                        Pattern pattern = Pattern.compile((String) rewrite.getKey());
+                        Matcher matcher = pattern.matcher(connection);
+                        if (matcher.matches()) {
+                            if (matcher.groupCount() > 0) {
+                                tag = matcher.group(1);
+                            }
+
+                            connection = matcher.replaceAll((String) rewrite.getValue());
+
+                            break dance;
```

Comment:
Break dance! 💯

Reply:
LOL this is an old joke since the College time :D